src/hdf5/osw_2ioda.py

```diff
@@ -70,11 +66,6 @@
 
 # Assign missing value details for the variables
 string_missing_value = '_'
```

Comment:
The `iconv.get_default_fill_val` function is listed below. It should be mimicking the netcdf default fill values which are good for a ioda file. The ioda reader will convert any specified fill values to the internal JEDI missing values, so might as well use the netcdf defaults. Here's the code:

https://github.com/JCSDA-internal/ioda-converters/blob/4af7be2f4a8fcdcf0e08b2518ff3c17775b4169d/src/pyiodaconv/ioda_conv_engines.py#L42-L69

It's not very clear but the np.S1 data type can be used for a string and you get an empty string back (\x00 -> null byte). Note that np.U1 or np.object can also be used to return a string containing a null byte. I believe that np.S1 is a 1 character string in ASCII, and np.U1 is a 1 character string in unicode, and np.object is a catch all type that we are using for datetime objects and if not a datetime, then it falls back on the null byte.

I think it would be good to continue using the `iconv.get_default_fill_val` function, and if you want it to take a python data type argument, then this function could be updated accordingly. We originally used numpy datatypes because all of the python converters were based on numpy arrays (which may not be true now). Also if you want to change the string default fill value to an underscore, that could be updated in the function too. Using the function helps keep things consistent across all the python converters.

Reply:
great thanks @srherbener think I will create a new issue and associated PR that adds all the types here:
https://github.com/JCSDA-internal/ioda-converters/blob/4af7be2f4a8fcdcf0e08b2518ff3c17775b4169d/src/pyiodaconv/def_jedi_utils.py#L22

it's missing `long` and will discuss with you about adding the null byte for missing string in that issue.  Also will then modify all the routines in this directory (and a few others) that are using this template

you're good to move forward with this PR correct?

Reply:
I think it makes sense to create the new issue, merge this PR in now and address the data type issues later. I'll hit the approve button.